src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-list/osd-list.component.ts

```diff
@@ -426,10 +429,13 @@ export class OsdListComponent extends ListWithDetails implements OnInit {
     }
   }
 
-  getOsdList() {
-    const observables = [this.osdService.getList(), this.osdService.getFlags()];
-    observableForkJoin(observables).subscribe((resp: [any[], string[]]) => {
-      this.osds = resp[0].map((osd) => {
+  getOsdList(context?: CdTableFetchDataContext) {
+    if (!context) context = new CdTableFetchDataContext(() => undefined);
```

Comment:
why this "() => undefined", I am not following, sorry.

Reply:
i just copied it from other parts but its basically passing an empty arrow function to the `CdTableFetchDataContext` because its constructor expects an error fn.

Reply:
I see this is only for error and to be called from `fetchData()` and that we do call it at [some places.](https://github.com/ceph/ceph/blob/main/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/services.component.ts#L247)

```
/**
   * The function that should be called from within the error handler
   * of the 'fetchData' function to display the error panel and to
   * reset the data table to the correct state.
   */
  error: Function;
  ```
Right now I dont see any instances of a custom error function in code.
As almost everywhere we require this to be unset `() => undefined`.

So we may want to revisit and think if its still required, only [Table Component ](https://github.com/ceph/ceph/blob/398563d959f96017fda47be5fac9d0a84c84ec1e/src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts#L844)sets the context with error function, so perhaps we can bring that here in `CdTableFetchDataContext` and avoid sending it in params.

Not blocking this PR on this one, but since it came up here, hence commented.

But for this PR and also at other places may be we can just do if(context) and then proceed, why need a dummy context ?

Reply:
if you check it in the osd-list.component, the `getOsdList()` is called in other places possibly to refresh the table after editing or updating some flags and during those calls, the context are not set. so if we just get the list only when we have `context` we'll be looking at an outdated data for these calls which is not what we need here (and certainly will be a regression).

maybe we look into making that error function an optional parameter in the constructor so we don't need to put undefined everytime, but shall we do it separately?

Reply:
for now I made the `error` optional in the `CdTableFetchDataContext` and removed the `undefined` function.